modin/numpy/arr.py

```diff
@@ -1390,6 +1390,71 @@ def __rmul__(
     ):
         return self.multiply(x2, out, where, casting, order, dtype, subok)
 
+    def dot(self, other, out=None):
+        other = try_convert_from_interoperable_type(other)
+        if numpy.isscalar(other):
+            # other is scalar -- result is an array
+            result = self._query_compiler.mul(other)
+            result_ndim = self._ndim
+        elif not isinstance(other, array):
+            raise TypeError(
+                f"Unsupported operand type(s): '{type(self)}' and '{type(other)}'"
+            )
+        elif self._ndim == 1 and other._ndim == 1:
+            # both 1D arrays -- result is a scalar
+            result = self._query_compiler.dot(
+                other._query_compiler, squeeze_self=True, squeeze_other=True
+            )
+            return result.to_numpy()[0, 0]
+        elif self._ndim == 2 and other._ndim == 2:
+            # both 2D arrays -- result is a 2D array
+            result = self._query_compiler.dot(other._query_compiler)
+            result_ndim = 2
+        elif self._ndim == 1 and other._ndim == 2:
+            result = self._query_compiler.dot(other._query_compiler, squeeze_self=True)
+            result_ndim = 1
+        elif self._ndim == 2 and other._ndim == 1:
+            result = self._query_compiler.dot(other._query_compiler)
+            result_ndim = 1
+        if out is not None:
+            out._update_inplace(result)
+        return array(_query_compiler=result, _ndim=result_ndim)
+
+    def __matmul__(self, other):
+        if numpy.isscalar(other):
+            # numpy's original error message is something cryptic about a gufunc signature
+            raise ValueError(
+                "cannot call matmul with a scalar argument (use np.dot instead)"
+            )
+        return self.dot(other)
+
+    def _norm(self, ord=None, axis=None, keepdims=False):
+        check_kwargs(keepdims=keepdims)
+        if ord is not None and ord not in ("fro",):  # , numpy.inf, -numpy.inf, 0):
+            raise NotImplementedError("unsupported ord argument for norm:", ord)
+        if isinstance(axis, int) and axis < 0:
+            axis = self._ndim + axis
```

Comment:
We can't do this, since we need the original axis in the error (I.e. if users passed in -3 for axis, it should error and say axis=-3 is invalid, not that axis =-1 is invalid)

Reply:
Fixed.